java/core/src/test/java/sleeper/core/util/ByteArrayTest.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2022-2025 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.core.util;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.TreeSet;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ByteArrayTest {
+
+    @Test
+    void shouldProvideByteArrayAsPrimitiveType() {
+        byte[] data = {1, 2, 3};
+        // When
+        ByteArray result = ByteArray.wrap(data);
+
+        //Then
+        assertThat(result.getArray()).isEqualTo(data);
+        assertThat(result.getLength()).isEqualTo(data.length);
+    }
+
+    @Test
+    void shouldCompareByteArraysCorrectly() {
```

Comment:
This test name is too vague, it suggests it's asserting more than it is. I think you wanted to assert that when we wrap two separate arrays containing the same data, they are equal?